Sources/NetworkProtection/NetworkProtectionDeviceManager.swift

```diff
@@ -138,15 +138,28 @@ public actor NetworkProtectionDeviceManager: NetworkProtectionDeviceManagement {
         if regenerateKey {
             keyPair = keyStore.newKeyPair()
         } else {
-            keyPair = keyStore.currentKeyPair() ?? keyStore.newKeyPair()
+            // Temporary code added on 2024-03-12 to fix a previous issue where users had a really long
+            // key expiration date.  We should remove this after a month or so.
+            if let existingKeyPair = keyStore.currentKeyPair(),
+               existingKeyPair.expirationDate > Date().addingTimeInterval(TimeInterval.day) {
+
+                keyPair = keyStore.newKeyPair()
```

Comment:
If the expiration date is over a day from now, we create a new key pair.  This is to fix users with long expiration dates... although this could be a good security measure in case users are manipulating expiration dates.